src/Elasticsearch.Net/Connection/HttpConnection.cs

```diff
@@ -248,6 +248,8 @@ protected virtual HttpMessageHandler CreateHttpClientHandler(RequestData request
 
 		protected virtual HttpRequestMessage CreateHttpRequestMessage(RequestData requestData)
 		{
+			requestData.HttpClientIdentifier = ConnectionInfo.UsingCurlHandler ? "c" : "s";
```

Comment:
I think we should do this work when instantiating the connection configuration. And have the connection configuration expose two default headers (async, non async).

`HttpClientIdentifier` could become a property all `IConnection`'s have to expose.